cms/release_calendar/models.py

```diff
@@ -0,0 +1,201 @@
+from typing import TYPE_CHECKING, Any, ClassVar
+
+from django.conf import settings
+from django.db import models
+from django.utils.functional import cached_property
+from django.utils.translation import gettext_lazy as _
+from modelcluster.fields import ParentalKey
+from wagtail.admin.panels import FieldPanel, FieldRowPanel, InlinePanel, MultiFieldPanel
+from wagtail.fields import RichTextField
+from wagtail.models import Orderable, Page
+
+from cms.core.fields import StreamField
+from cms.core.models import BasePage, LinkFieldsMixin
+
+from .blocks import (
+    ReleaseCalendarChangesStoryBlock,
+    ReleaseCalendarPreReleaseAccessStoryBlock,
+    ReleaseCalendarStoryBlock,
+)
+from .enums import NON_PROVISIONAL_STATUSES, ReleaseStatus
+from .forms import ReleaseCalendarPageAdminForm
+
+if TYPE_CHECKING:
+    from django.http import HttpRequest
+
+
+class ReleaseCalendarIndex(BasePage):  # type: ignore[django-manager-missing]
+    """The release calendar index page placeholder."""
+
+    template = "templates/pages/release_index.html"
+
+    parent_page_types: ClassVar[list[str]] = ["home.HomePage"]
+    subpage_types: ClassVar[list[str]] = ["ReleaseCalendarPage"]
+    max_count_per_parent = 1
+
+
+class ReleasePageRelatedLink(LinkFieldsMixin, Orderable):
+    """Related links. e.g. https://www.ons.gov.uk/releases/welshlanguagecensus2021inwales."""
+
+    parent = ParentalKey("ReleaseCalendarPage", related_name="related_links", on_delete=models.CASCADE)
+
+
+class ReleaseCalendarPage(BasePage):  # type: ignore[django-manager-missing]
```

Comment:
Currently, this page can be deleted; we should prevent that. They should be able to cancel but not delete.

Reply:
How is 461eb36?

Reply:
That's great; I recommend we do the same for the "Release calendar index", which was programmatically created.
I haven't tested Move, but I suspect Move will enforce the subpage/parent page types list.

Reply:
Move indeed enforces parent/subpage types